lib/dry/result_matcher/evaluator.rb

```diff
@@ -0,0 +1,38 @@
+module Dry
+  module ResultMatcher
+    class Evaluator < BasicObject
+      def initialize(result, cases)
+        @cases = cases
+        @result = result
+        @matched = false
+        @output = nil
+      end
+
+      def call(&block)
+        block.call(self)
```

Comment:
Ah, thanks for the note here @flash-gordon! I'll make this change in master in a moment.